pkg/cloudprovider/azure.go

```diff
@@ -1,33 +1,121 @@
 package cloudprovider
 
 import (
+	"bytes"
+	"context"
+	"strings"
+
+	storagemgmt "github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-02-01/storage"
+	azstorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/google/uuid"
 	velero "github.com/heptio/velero/pkg/apis/velero/v1"
+	"github.com/joho/godotenv"
 	appsv1 "github.com/openshift/api/apps/v1"
+	"github.com/pkg/errors"
 	kapi "k8s.io/api/core/v1"
 )
 
+// Credentials secret.
+const (
+	AzureCredentials = "azure-credentials"
+
+	tenantIDEnvVar       = "AZURE_TENANT_ID"
+	subscriptionIDEnvVar = "AZURE_SUBSCRIPTION_ID"
+	clientIDEnvVar       = "AZURE_CLIENT_ID"
+	clientSecretEnvVar   = "AZURE_CLIENT_SECRET"
+	cloudNameEnvVar      = "AZURE_CLOUD_NAME"
```

Comment:
Are these really environment variables? ^^

Reply:
The secret data containing these vars is in the `.env` file format, which looks like this:
```
AZURE_TENANT_ID=foo
AZURE_SUBSCRIPTION_ID=foo
AZURE_CLIENT_ID=foo
```

This is the format in which Velero accepts credentials. I'm open to suggestions for naming these items something else, they're the keys within an environment variable `.env` file.

Reply:
IMHO, just dropping the `EnvVar` suffix would be better since the keys are used in both the secret and the azure .env file.  I found this confusing (at first) because it suggested (to me) actual  process environment variables.

Reply:
Replaced `EnvVar` with `Key`